File: sdk/runanywhere-swift/Sources/RunAnywhere/Infrastructure/FileManagement/Constants/StorageConstants.swift
import Foundation

/// Storage and file management configuration constants
public enum StorageConstants {

    // MARK: - Directory Names

    /// Model directory name
    public static let modelDirectoryName = "RunAnywhereModels"

    /// Cache directory name
    public static let cacheDirectoryName = "RunAnywhereCache"

    /// Temporary directory name
    public static let tempDirectoryName = "RunAnywhereTmp"

    // MARK: - Cache Configuration

    /// Cache size limit in bytes (100 MB)
    public static let cacheSizeLimit: Int64 = 100 * 1024 * 1024

    /// Default max cache size in MB
    public static let defaultMaxCacheSizeMB = 2048

    /// Default cleanup threshold percentage
    public static let defaultCleanupThresholdPercentage = 90

    /// Default model retention days
    public static let defaultModelRetentionDays = 30
}

File: sdk/runanywhere-react-native/src/Infrastructure/FileManagement/Constants/StorageConstants.ts
/**
 * StorageConstants.ts
 * RunAnywhere SDK
 *
 * Storage and file management configuration constants.
 * Matches iOS: Infrastructure/FileManagement/Constants/StorageConstants.swift
 */

/**
 * Storage and file management configuration constants
 */
export const StorageConstants = {
  // MARK: - Directory Names

  /** Model directory name */
  modelDirectoryName: 'RunAnywhereModels',

  /** Cache directory name */
  cacheDirectoryName: 'RunAnywhereCache',

  /** Temporary directory name */
  tempDirectoryName: 'RunAnywhereTmp',

  // MARK: - Cache Configuration

  /** Cache size limit in bytes (100 MB) */
  cacheSizeLimit: 100 * 1024 * 1024,

  /** Default max cache size in MB */
  defaultMaxCacheSizeMB: 2048,

  /** Default cleanup threshold percentage */
  defaultCleanupThresholdPercentage: 90,

  /** Default model retention days */
  defaultModelRetentionDays: 30,
} as const;

/**
 * Type for StorageConstants
 */
export type StorageConstantsType = typeof StorageConstants;
